RoR2BepInExPack/VanillaFixes/FixProjectileCatalogLimitError.cs

```diff
@@ -0,0 +1,65 @@
+
+using MonoMod.RuntimeDetour;
+using MonoMod.Cil;
+using EntityStates;
+using RoR2BepInExPack.Reflection;
+using System;
+using Mono.Cecil.Cil;
+using UnityEngine;
+using RoR2.Items;
+using System.Reflection;
+using RoR2;
+
+namespace RoR2BepInExPack.VanillaFixes;
+
+// The ProjectileCatalog logs an error if more than 256 projectiles are registered, despite the actual limit being much higher.
+// Fix: Update the projectile limit to prevent the misleading error
+internal class FixProjectileCatalogLimitError
+{
+    private static ILHook _ilHook;
+
+    internal static void Init()
+    {
+        var ilHookConfig = new ILHookConfig() { ManualApply = true };
+        _ilHook = new ILHook(
+                    typeof(ProjectileCatalog).GetMethod(nameof(ProjectileCatalog.SetProjectilePrefabs), ReflectionHelper.AllFlags),
+                    IncreaseCatalogLimit,
+                    ref ilHookConfig
+                );
+    }
+    internal static void Enable()
+    {
+        _ilHook.Apply();
+    }
+    internal static void Disable()
+    {
+        _ilHook.Undo();
+    }
+    internal static void Destroy()
+    {
+        _ilHook.Free();
+    }
+    private static void IncreaseCatalogLimit(ILContext il)
+    {
+        ILCursor c = new ILCursor(il);
+        int locLimitIndex = -1;
+        bool ILFound = c.TryGotoNext(MoveType.Before,
+            x => x.MatchLdcI4(out _),
+            x => x.MatchStloc(out locLimitIndex),
+            x => x.MatchLdsfld(typeof(ProjectileCatalog).GetField(nameof(ProjectileCatalog.projectilePrefabs), ReflectionHelper.AllFlags)),
+            x => x.MatchLdlen(),
+            x => x.MatchConvI4(),
+            x => x.MatchLdloc(locLimitIndex),
+            x => x.MatchBle(out _)
+            );
+
+        if (ILFound)
+        {
+            c.Next.Operand = int.MaxValue;
```

Comment:
Wouldn't it be better to insert an unconditional jump instead?

Reply:
Not sure what you mean by "better" exactly, this seems like a reasonable way to implement the fix while respecting the vanilla code to me but I guess I can change it if you like

Reply:
Either way is functional so "better" might be the wrong word to use.I've just anectodally found IL manipulation that doesn't change existing instructions to be more robust.
Something could also be said about ignoring the work done by the check but that is too small an impact to be truly relevant here.